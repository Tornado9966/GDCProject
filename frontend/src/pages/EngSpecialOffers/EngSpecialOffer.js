import React from 'react';
import {Image, Button} from 'semantic-ui-react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import {close, seeMore, defaultTitle} from 'constants/text-constants';

import defaultImg from './default.jpg';
 

class EngSpecialOffer extends React.Component {
    static propTypes = {
        imgUrl: PropTypes.string,  
        title: PropTypes.string, 
        summary: PropTypes.string.isRequired,
        details: PropTypes.string.isRequired
    };
    
    static defaultProps = {
        imgUrl: defaultImg,  
        title: defaultTitle
    };

     state = {
        isToggleOn: false,
    }
    handleClick = () => {
        this.setState(state => ({
          isToggleOn: !state.isToggleOn
        }));
    }
    render() {
        const {imgUrl, title, summary, details} = this.props;
        const {isToggleOn} = this.state;
        const detailsClass = classnames('details-block', {
            show: isToggleOn,
            hide: !isToggleOn
          });

            return (
                <div className='specialitem'>
                    <Image 
                        src={imgUrl} 
                        size='large' 
                        floated='left' 
                    />
                    <div className='specialitem__info'>
                        <h2>{title}</h2>
                        <p>{summary}</p>
                        <p className={detailsClass}>{details}</p>
                        <Button className='specialitem__button' onClick={this.handleClick}>
                            {this.state.isToggleOn ? close : seeMore }
                        </Button>
                    </div>
                </div>
        );
    }
}

export default EngSpecialOffer;
