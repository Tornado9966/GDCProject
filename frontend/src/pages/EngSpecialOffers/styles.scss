@import 'assets/_colors.scss';

.main__special {
    width: 80%;
    margin: 100px auto;
    display: flex;
    flex-direction: column;
}

.specialitem {
    border-bottom: 1px solid $grey;
    margin-bottom: 60px;
    padding-bottom: 45px;
    display: flex;
    align-items: center;

    &:first-child {
        margin-top: 20px;
    }
    &:last-child {
        border-bottom: none;
        margin-bottom: 0;
    }

    img {
        &.image {
            &.large {
                width: 35vw;
                margin: 0;
            }
        
        }
    }
}

.specialitem__info {
    display: flex;
    flex-direction: column;
    padding-left: 50px;

    h2 {
        text-transform: uppercase;
        margin-bottom: 1.5vw;
        font-size: 1.6vw;
    }

    p {
        font-size: 1vw;
        line-height: 1.8vw;
        margin-bottom: 1vw;

        &.details-block {
            &.hide {
                max-height: 0;
                overflow: hidden;
                opacity: 0;
                transition: all 0.4s ease-out;
                margin: 0;
            }
            &.show {
                max-height: 500px;
                opacity: 1;
                overflow: hidden;
                transition: all 0.4s ease-in;
                margin: 0;
            }
        }
    }

    .specialitem__button {
        width: 15vw;
        border: 2px solid $button_color;
        color:  $button_color;
        background: transparent;
        transition: all 0.3s linear;
        margin-top: 20px;
        box-shadow: none;
    
        &:hover, &:active, &:focus {
            color: $white;
            background: $button_color;
            transition: all 0.3s linear;
            box-shadow: none;
        }
    }

}

@media all and (max-width: 992px) {
    .main__special {
        margin: 1% auto;
        width: 95%;
    }
    .specialitem {
        flex-direction: column;
        padding-bottom: 1%;
        margin-bottom: 2%;
        margin-top: 3%;
        &:last-child {
            margin-bottom: 0;
        }
        img {
            &.image {
                &.large {
                    width: 60vw;
                }
            }
        }
    }
    .specialitem__info {
        width: 77%;
        margin: 2% auto;
        padding: 0;
        h2 {
            margin-top: 1.5vw;
            font-size: 3.4vw;
            text-align: center;
        }
        p {
            font-size: 2.2vw;
            line-height: 2.6vw;
            text-align: justify;
        }
        .specialitem__button {
            width: 17vw;
            height: 7vw;
            margin: 3% auto;
            font-size: 2vw;
        }

    }
}


