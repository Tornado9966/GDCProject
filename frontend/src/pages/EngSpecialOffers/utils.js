import {getSpecialOffers} from '../../services/http-client/special-offers';

export function getSpecialOffersList(count) {
    return getSpecialOffers(count);
}
