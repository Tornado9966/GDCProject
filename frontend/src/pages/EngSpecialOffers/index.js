import {EngSpecialOffers} from './EngSpecialOffers';

export default EngSpecialOffers;
