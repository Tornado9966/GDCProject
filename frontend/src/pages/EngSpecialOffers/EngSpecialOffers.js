import React from 'react';
import {map as _map} from 'lodash';
import PropTypes from 'prop-types'; 

import EngSpecialOffer from './EngSpecialOffer';

import {getSpecialOffersList} from './utils';

import './styles.scss';

export class EngSpecialOffers extends React.Component {
    static propTypes = {
        count: PropTypes.number,
    };

    static defaultProps = {
        count: 4,
    };

    state = {
        specialOffers: [],
    };

    componentDidMount() {
        return getSpecialOffersList(this.props.count)
            .then(res => this.setState({specialOffers: res.data}) );
    }

    render() {
        return(
            <div className='main__special'>
                {_map(this.state.specialOffers, ({_id, imgUrl, title, summary, details}) => 
                <EngSpecialOffer
                    key={_id}
                    imgUrl={imgUrl} 
                    title={title}
                    summary={summary}
                    details={details} 
                />)}
            </div>
        );
    }
    
}
