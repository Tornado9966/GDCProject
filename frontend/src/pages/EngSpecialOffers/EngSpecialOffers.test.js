import React from 'react';
import {shallow, mount} from 'enzyme';

import {EngSpecialOffers} from './EngSpecialOffers';

const setUp = (props = {}) => {
    const component = shallow(<EngSpecialOffers {...props} />);
    return component;
};
describe('EngSpecialOffers', () => {
    let component;
    const offers = {
        imgUrl: 'https://firebasestorage.googleapis.com/v0/b/eatngo-72705.appspot.com/o/breakfast.jpg?alt=media&token=776476df-1efb-43ba-9a0f-a3bf5938c155',
        title: 'Discount',
        summary: 'Discount',
        details: 'More about discount'
    };
    beforeEach(() => {
        component = setUp();
    });
    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
      });    
    it('render component EngSpecialOffers', () => {
        const component = shallow(<EngSpecialOffers />);
        expect(component).toHaveLength(1);
    });
    it('should exist' ,() => {
        const component = mount(<EngSpecialOffers offers={offers} />);
        expect(component.props().offers).toEqual(offers);
   });
});