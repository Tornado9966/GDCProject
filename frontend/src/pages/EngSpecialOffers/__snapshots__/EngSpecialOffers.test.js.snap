// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngSpecialOffers renders correctly 1`] = `undefined`;
